homeassistant/components/hvv_departures/sensor.py

```diff
@@ -0,0 +1,215 @@
+"""Sensor platform for hvv."""
+from datetime import datetime, timedelta
+import logging
+
+from aiohttp import ClientConnectorError
+from pygti.exceptions import InvalidAuth
+from pygti.gti import GTI, Auth
+
+from homeassistant.const import DEVICE_CLASS_TIMESTAMP
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+from .const import ATTRIBUTION, DOMAIN, MANUFACTURER
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+MAX_LIST = 20
+MAX_TIME_OFFSET = 360
+ICON = "mdi:bus"
+UNIT_OF_MEASUREMENT = "min"
+
+ATTR_ATTRIBUTION = "attribution"
+ATTR_DEPARTURE = "departure"
+ATTR_LINE = "line"
+ATTR_ORIGIN = "origin"
+ATTR_DIRECTION = "direction"
+ATTR_TYPE = "type"
+ATTR_ID = "id"
+ATTR_DELAY = "delay"
+ATTR_NEXT = "next"
+
+PARALLEL_UPDATES = 0
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass, config, async_add_entities, discovery_info=None
+):  # pylint: disable=unused-argument
+    """Set up the sensor platform.
+
+    Skipped, as setup through configuration.yaml is not supported.
+    """
+
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the sensor platform."""
+
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    sensor = HVVDepartureSensor(hass, config_entry, session)
+    async_add_devices([sensor], True)
+
+
+class HVVDepartureSensor(Entity):
+    """HVVDepartureSensor class."""
+
+    def __init__(self, hass, config_entry, session):
+        """Initialize."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.station_name = self.config_entry.data["station"]["name"]
+        self.attr = {ATTR_ATTRIBUTION: ATTRIBUTION}
+        self._available = False
+        self._state = None
+        self._name = f"Departures at {self.station_name}"
+        self._last_error = None
+
+        self.gti = GTI(
+            Auth(
+                session,
+                self.config_entry.data["username"],
+                self.config_entry.data["password"],
+                self.config_entry.data["host"],
+            )
+        )
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self, **kwargs):
+        """Update the sensor."""
+
+        try:
```

Comment:
Only wrap the line where an error could occur.

Reply:
This has not been resolved.

Reply:
Understood it wrong, _should_ be correct now